### PR TITLE
chore: consolidate design proposals

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- Removes the design folder from `contribute`.
- Removes the deprecated engines related design.
- Places the design for the all engines under `design`.
- Updates the README.
- Updates the oep-template and oep-process.

Addesses - #3744

### CODE REVIEW

File: designs/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+## Common or Generic 
+
+- [openebs/openebs](https://github.com/openebs/openebs) is OpenEBS meta repository that contains design documents, project management, community and contributor documents, deployment and workload examples. 
+- [openebs/website](https://github.com/openebs/website) contains the source code for OpenEBS portal (https://openebs.io) implemented using Gatsby framework and other libraries listed in [package.json](https://github.com/openebs/website/blob/refactor-to-ghost-and-gatsby/package.json).
+- [openebs/velero-plugin](https://github.com/openebs/velero-plugin) has the plugin code to perform cStor and ZFS Local PV based Backup and Restore using Velero. 
+- [openebs/linux-utils](https://github.com/openebs/linux-utils) is a general purpose alpine container used to launch some management jobs by OpenEBS Operators. 
+- [openebs/openebsctl](https://github.com/openebs/openebsctl) contains the OpenEBS CLI that can be run as kubectl plugin. This functionality is being split from the mono-repo [openebs/maya](https://github.com/openebs/maya) into its own repository. (Currently in Alpha). 
+
+## Experimental repos
```
Comment (avishnu) [inline designs/README.md:44]: This section can be moved below section for Local PVs?

File: designs/local-pv/hostpath/design/hostpath_localpv_provisioner.md
Comment (niladrih) [inline designs/local-pv/hostpath/design/hostpath_localpv_provisioner.md:1]: designs/local-pv/hostpath/design/hostpath_localpv_provisioner.md --> designs/local-pv/hostpath/hostpath_localpv_provisioner.md

File: designs/local-pv/hostpath/design/images/hostpath_localpv_provisioner-deployment.jpg
Comment (niladrih) [inline designs/local-pv/hostpath/design/images/hostpath_localpv_provisioner-deployment.jpg:1]: designs/local-pv/hostpath/design/images/hostpath_localpv_provisioner-deployment.jpg --> designs/local-pv/hostpath/images/hostpath_localpv_provisioner-deployment.jpg

**Comment by niladrih** [comment]
If we're going to continue with the 'oep' name, then maybe this is better directory structure:
```
oep/
├── oep-process.md
├── oep-template.md
└── designs/
    ├── hostpath
    ├── lvm
    └── zfs
```

**Comment by avishnu** [comment]
> If we're going to continue with the 'oep' name, then maybe this is better directory structure:
> 
> ```
> oep/
> ├── oep-process.md
> ├── oep-template.md
> └── designs/
>     └── local-pv/
>         ├── hostpath
>         ├── lvm
>         └── zfs
> ```

I think we should stick to 'designs' as the first-level directory.
oep-template and oep-process can get inside the contribution section? or inside the .github?

**Comment by niladrih** [comment]
> oep-template and oep-process can get inside the contribution section? or inside the .github?
Sure, they could be linked on the CONTRIBUTING.md. Not sure if they should live inside .github, like issue_template.md files. Issues are a github-specific item, so might make sense for them to live inside .github. Not sure about the oep-template.

**Comment by avishnu** [comment]
> > oep-template and oep-process can get inside the contribution section? or inside the .github?
> 
> Sure, they could be linked on the CONTRIBUTING.md. Not sure if they should live inside .github, like issue_template.md files. Issues are a github-specific item, so might make sense for them to live inside .github. Not sure about the oep-template.

If all the designs are following the same format as outlined in oep-template.md and oep-process.md, then we could have the 2 files placed directly under 'designs' directory IMO.

**Comment by Abhinandan-Purkait** [comment]
> > > oep-template and oep-process can get inside the contribution section? or inside the .github?
> > 
> > 
> > Sure, they could be linked on the CONTRIBUTING.md. Not sure if they should live inside .github, like issue_template.md files. Issues are a github-specific item, so might make sense for them to live inside .github. Not sure about the oep-template.
> 
> If all the designs are following the same format as outlined in oep-template.md and oep-process.md, then we could have the 2 files placed directly under 'designs' directory IMO.

I don't think we would restrict anyone to use that. It's just a way to convey what all items can be a part of a design proposal. If we want to restrict it we can keep it here. Otherwise we can place it somewhere else and add links in the README.

**Review by avishnu — APPROVED** [review]

**Review by niladrih — APPROVED** [review]

**Pull request merged**